diff -rupN a/src/video/android/SDL_androidevents.c b/src/video/android/SDL_androidevents.c
--- a/src/video/android/SDL_androidevents.c
+++ b/src/video/android/SDL_androidevents.c
@@ -121 +121 @@ void Android_PumpEvents_Blocking(_THIS)
-            android_egl_context_backup(Android_Window);
+            /* android_egl_context_backup(Android_Window); */
@@ -147 +147 @@ void Android_PumpEvents_Blocking(_THIS)
-                android_egl_context_restore(Android_Window);
+                /* android_egl_context_restore(Android_Window); */
@@ -196 +196 @@ void Android_PumpEvents_NonBlocking(_THIS)
-                android_egl_context_backup(Android_Window);
+                /* android_egl_context_backup(Android_Window); */
@@ -229 +229 @@ void Android_PumpEvents_NonBlocking(_THIS)
-                android_egl_context_restore(Android_Window);
+                /* android_egl_context_restore(Android_Window); */
