diff -rupN a/src/events/SDL_events.c b/src/events/SDL_events.c
--- a/src/events/SDL_events.c
+++ b/src/events/SDL_events.c
@@ -35,0 +36,2 @@
+#include <pthread.h>
+
@@ -922 +924 @@ static void SDL_PumpEventsInternal(SDL_bool push_sentinel)
-    if (SDL_update_joysticks) {
+    if (pthread_main_np() && SDL_update_joysticks) {
diff -rupN a/src/video/cocoa/SDL_cocoaevents.m b/src/video/cocoa/SDL_cocoaevents.m
--- a/src/video/cocoa/SDL_cocoaevents.m
+++ b/src/video/cocoa/SDL_cocoaevents.m
@@ -29,0 +30,2 @@
+#include <pthread.h>
+
@@ -572 +574,3 @@ void Cocoa_PumpEvents(_THIS)
-    Cocoa_PumpEventsUntilDate(_this, [NSDate distantPast], true);
+    if (pthread_main_np()) {
+        Cocoa_PumpEventsUntilDate(_this, [NSDate distantPast], true);
+    }
